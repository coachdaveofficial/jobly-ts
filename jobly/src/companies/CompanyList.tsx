import React, { useState, useEffect } from 'react'
import { Company } from '../types/types';
import JoblyApi from '../api'
import CompanyCard from './CompanyCard';

/**
 * Renders a list of all companies 
 */
export default function CompanyList() {
    const [companies, setCompanies] = useState<Company[]>([]);
    useEffect(function getCompaniesOnMount() {
        search(undefined);
    }, []);

    /** Triggered by search form submit; reloads companies. */
    async function search(name:string | undefined) {
        let companies: Company[] = await JoblyApi.getCompanies(name);
        setCompanies(companies);
    }

    if (!companies) return <div>Loading...</div>

    return (
        <div className='col-md-8 offset-md-2'>
            {companies.map((c) => (
                <CompanyCard 
                    key={c.handle}
                    name={c.name}
                    description={c.description}
                    logoUrl={c.logoUrl} 
                    handle={c.handle} 
                    numEmployees={c.numEmployees} 
                    jobs={c.jobs}                   
                    />

            ))}
        </div>
    );
}