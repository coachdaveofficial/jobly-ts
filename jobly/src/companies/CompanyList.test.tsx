import React from "react";
import { act, render, waitFor, screen } from "@testing-library/react";
import { MemoryRouter, useParams } from "react-router-dom";
import CompanyList from "./CompanyList";
import JoblyApi from "../api";

it("matches the snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyList />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

const handle = 'example-handle';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useParams: jest.fn(),
}));

jest.mock('../api', () => ({
    getCompanies: jest.fn(),
}));

const mockCompany = {
    name: 'Example Company',
    description: 'Lorem ipsum dolor sit amet',
    jobs: [
        { id: 1, title: 'Job 1', salary: 50000, equity: 0.2, companyHandle: 'example-handle' },
        { id: 2, title: 'Job 2', salary: 60000, equity: 0.3, companyHandle: 'example-handle' },
    ],
};

/*
test('renders the company details', async () => {
    (JoblyApi.getCompanies as jest.Mock).mockResolvedValue(mockCompany);
  
    await act(async () => {
      const { getByText, findByText } = render(<CompanyList />);
  
      // Wait for the component to finish loading
      await findByText(mockCompany.name);
  
      expect(findByText(mockCompany.name)).toBeInTheDocument();
      expect(getByText(mockCompany.name)).toBeInTheDocument();
      expect(getByText(mockCompany.description)).toBeInTheDocument();
      expect(getByText('Job 1')).toBeInTheDocument();
      expect(getByText('Job 2')).toBeInTheDocument();
    });
});
*/
