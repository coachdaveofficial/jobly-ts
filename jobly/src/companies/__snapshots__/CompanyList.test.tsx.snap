// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-8 offset-md-2"
  >
    <div
      class="SearchForm mb-4"
    >
      <form
        class="form-inline"
      >
        <input
          class="form-control form-control-lg flex-grow-1"
          name="searchTerm"
          placeholder="Enter search term..."
          value=""
        />
        <button
          class="btn btn-lg btn-primary"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
